### PR TITLE
New rules, bump deps, output

### PR DESCRIPTION
**Pull request by scovetta**
This PR adds a few new backdoor rules, bumps dependencies (AppInspector in particular), and improves the output.

### CODE REVIEW

**Comment by scovetta** [comment]
I think we'll have some conflicts here, feel free to trash mine in favor of yours, @gfs.

File: src/oss-characteristics/CharacteristicTool.cs [hunk 1]
```diff
@@ -310,6 +326,42 @@ public async Task RunAsync(Options options)
                                 var encodedName = Uri.EscapeUriString(target ?? "unknown");
                                 var purl = new PackageURL("generic", encodedName);
                                 AppendOutput(outputBuilder, purl, analysisResults);
+                                finalResults.Add(analysisResults);
+                            }
+                        }
+                        else if (File.Exists(target))
```
Comment (gfs) [inline src/oss-characteristics/CharacteristicTool.cs:332]: AppInspector can take a file or archive as argument you don’t need to pre extract and then delete.

**Comment by gfs** [comment]
> I think we'll have some conflicts here, feel free to trash mine in favor of yours, @gfs.

I’ll figure out the right way to merge on Monday. Probably will manually pick your changes into mine.

File: src/oss-characteristics/CharacteristicTool.cs [hunk 2]
```diff
@@ -310,6 +326,42 @@ public async Task RunAsync(Options options)
...
+                        {
+                            Extractor e = new Extractor();
+                            var tempDirectory = Path.Join(Path.GetTempPath(), Guid.NewGuid().ToString());
+                            var extractorOptions = new ExtractorOptions()
+                            {
+                                ExtractSelfOnFail = true,
```
Comment (gfs) [inline src/oss-characteristics/CharacteristicTool.cs:338]: @scovetta FYI on this behavior. This means if we think it’s an archive and it fails to extract we return the archive itself. That means that it is likely a binary file based on the header. Not sure what AI does with those or if that’s the intention here. If you pass RE a regular non archive file you just get it back in a stream regardless of this option.

Comment (scovetta) [reply src/oss-characteristics/CharacteristicTool.cs:338]: Got it, thanks for the clarification!

**Comment by scovetta** [comment]
> > I think we'll have some conflicts here, feel free to trash mine in favor of yours, @gfs.
> 
> I’ll figure out the right way to merge on Monday. Probably will manually pick your changes into mine.

Sounds good, I won't make any more structural changes -- maybe a few rule updates, but that's it, for the next couple days.

**Comment by gfs** [comment]
> > > I think we'll have some conflicts here, feel free to trash mine in favor of yours, @gfs.
> > 
> > 
> > I’ll figure out the right way to merge on Monday. Probably will manually pick your changes into mine.
> 
> Sounds good, I won't make any more structural changes -- maybe a few rule updates, but that's it, for the next couple days.

Feel free to make rule updates directly in my PR

**Comment by gfs** [comment]
folded these changes into #210

**Pull request closed without merging**